### PR TITLE
feat(ui): update status labels to follow consistent design pattern

### PR DESCRIPTION
**Pull request by rsun19**
## Description

Updates status labels across the Model Registry UI:

01-model-registry-settings
02-model-transfer-jobs
03-transfer-job-status-modal
04-catalog-source-settings
05-archived-model-details

**Styling convention:**
- **Interactive labels** (clicking opens a popover or modal) use `variant="filled"`
- **Non-interactive labels** use `variant="outline"`

**Model Registry Settings (`ModelRegistryTableRowStatus`):**
- Renamed status labels: `Available` → `Ready`, `Progressing` → `Starting`
- Added new `Stopping` status (triggered when `metadata.deletionTimestamp` is set)
- Applied filled/outline variant based on interactivity

**Model Transfer Jobs (`ModelTransferJobTableRow` / `ModelTransferJobStatusModal`):**
- Changed `Pending` label color from grey → purple
- Changed `Cancelled` label text to `Canceled`
- Added explicit `variant="filled"` on interactive table row label
- Added `variant="outline"` on non-interactive modal label

**Catalog Source Settings (`CatalogSourceStatus` / `CatalogSourceStatusErrorModal`):**
- Added `variant="outline"` to all non-interactive status labels: Connected, Failed, Starting, Unknown

**Archive Details (`RegisteredModelArchiveDetails`, `ModelVersionArchiveDetails`, `ArchiveModelVersionDetails`):**
- Added `variant="outline"` to non-interactive "Archived" labels

**BFF Mock Data (`model_registry_settings_handler.go`):**
- Added `DeletionTimestamp` field to `Metadata` struct
- Extended `createSampleModelRegistry` to accept `deletionTimestamp` and `conditions` parameters
- Added mock registries for each status (Ready, Starting, Stopping, Degrading, Unavailable) alongside existing mocks

## How Has This Been Tested?

- Added unit tests for `ModelRegistryTableRowStatus`: Stopping status, deletionTimestamp priority, and variant assertions (outline for non-interactive, filled for interactive)
- Added unit tests for `getStatusLabel`: all 5 transfer job statuses with correct labels and colors
- Added unit test for `ModelTransferJobStatusModal`: outline variant assertion
- Created new `CatalogSourceStatus.spec.tsx` test suite: variant assertions for Connected, Failed, Starting, Unknown labels plus default/disabled source handling
- Updated Cypress test assertion for `Cancelled` → `Canceled`
- All 32 unit tests pass

## Merge criteria:
- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits) (To pass the `DCO` check)

- [x] The commits have meaningful messages
- [x] Automated tests are provided as part of the PR for major new functionalities; testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).
- [ ] **For first time contributors**: Please reach out to the [Reviewers](https://github.com/kubeflow/model-registry/blob/main/OWNERS) to ensure all tests are being run, ensuring the label `ok-to-test` has been added to the PR.

If you have UI changes

- [x] The developer has added tests or explained why testing cannot be added.
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/frontend/src/app/pages/settings/ModelRegistriesTableRow.tsx [hunk 1]
```diff
@@ -43,7 +43,10 @@ const ModelRegistriesTableRow: React.FC<ModelRegistriesTableRowProps> = ({
         )}
       </Td>
       <Td dataLabel="Status" style={{ verticalAlign: 'middle' }}>
-        <ModelRegistryTableRowStatus conditions={mr.status?.conditions} />
+        <ModelRegistryTableRowStatus
+          conditions={mr.status?.conditions}
+          deletionTimestamp={mr.metadata.deletionTimestamp}
```
Comment (manaswinidas) [inline clients/ui/frontend/src/app/pages/settings/ModelRegistriesTableRow.tsx:48]: Does the `ModelRegistryKind` type from `mod-arch-shared` actually include `deletionTimestamp` on its metadata? If not, this silently passes `undefined` and the "Stopping" status becomes dead code. Please verify the shared type has this field — otherwise the BFF model was updated but the frontend type contract doesn't match.

File: clients/ui/frontend/src/app/pages/settings/__tests__/ModelRegistryTableRowStatus.spec.tsx [hunk 2]
```diff
@@ -167,7 +167,7 @@ describe('ModelRegistryTableRowStatus', () => {
         ]}
       />,
     );
-    expect(screen.getByText('Available')).toBeVisible();
+    expect(screen.getByText('Ready')).toBeVisible();
   });
   it('renders "Progressing" status', async () => {
```
Comment (manaswinidas) [inline clients/ui/frontend/src/app/pages/settings/__tests__/ModelRegistryTableRowStatus.spec.tsx]: This test is titled `"Progressing"` but actually asserts `screen.getByText('Unavailable')` (line 191). Since this PR renames all the other test titles to match the new label names (Available → Ready, Progressing → Starting), this pre-existing mislabel should be fixed here too — otherwise it's the only test title that doesn't match what it verifies.

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { ModelTransferJobStatus } from '~/app/types';
+import { getStatusLabel } from '~/app/pages/modelRegistry/screens/ModelTransferJobs/ModelTransferJobTableRow';
+
+describe('getStatusLabel', () => {
```
Comment (manaswinidas) [inline clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts:4]: Nit: These 5 tests mirror the switch cases 1:1 without testing edge cases (default branch, `icon` return value). Consider adding a test for an unknown/unexpected status value to cover the default branch, and optionally asserting `icon` — otherwise these are just restating the implementation as assertions.

**Review by manaswinidas — COMMENTED** [review]
Nice cleanup overall — the filled/outline convention and label renames look good. Left a couple of inline comments: one potential issue with the shared type for `deletionTimestamp`, one stale test name, and a nit on test coverage.

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { ModelTransferJobStatus } from '~/app/types';
```
Comment (manaswinidas) [inline clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts:1]: This can be removed mostly

File: clients/ui/frontend/src/app/pages/settings/ModelRegistriesTableRow.tsx (hunk 1 above)
Comment (rsun19) [reply clients/ui/frontend/src/app/pages/settings/ModelRegistriesTableRow.tsx:48]: Verified — `ModelRegistryKind` is `K8sResourceCommon & { metadata: { name, namespace, ... } }`. The `K8sResourceCommon` type (from `mod-arch-shared@1.12.0` in `dist/types/common.d.ts`) defines `metadata` as `Partial<{ ..., deletionTimestamp: string, ... }>`. TypeScript intersects both metadata shapes, so `deletionTimestamp` is available as an optional `string`. The Go BFF also includes it as `*time.Time` with `omitempty`, so the field is only present in the JSON when the resource is being deleted.

Comment (rsun19) [reply clients/ui/frontend/src/app/pages/settings/ModelRegistriesTableRow.tsx:48]: https://github.com/opendatahub-io/mod-arch-library/blob/main/mod-arch-shared/types/common.ts#L24

File: clients/ui/frontend/src/app/pages/settings/__tests__/ModelRegistryTableRowStatus.spec.tsx (hunk 2 above)
Comment (rsun19) [reply clients/ui/frontend/src/app/pages/settings/__tests__/ModelRegistryTableRowStatus.spec.tsx]: Good catch — fixed. Renamed the test title from `"Progressing"` to `"Unavailable"` to match the actual assertion.

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts (hunk 3 above)
Comment (rsun19) [reply clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts:4]: Done — consolidated the 5 individual tests into a single `it.each` table, added `icon` assertion (`toBeDefined()`), and added a new test for an unknown status value that covers the default branch (asserts raw status string, grey color, and null icon).

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts (hunk 4 above)
Comment (rsun19) [reply clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelTransferJobs/__tests__/getStatusLabel.spec.ts:1]: Refactored — consolidated the verbose per-status tests into a compact `it.each` table and added a default-branch edge case test. Much leaner now.

**Comment by rsun19** [comment]
In `useModelTransferJobs.spec.ts`, I also fixed this warning: `A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks. Active timers can also cause this, ensure that .unref() was called on them.`

**Comment by Misjohns** [comment]
@rsun19 I was out on PTO last week. Took a look at the screen caps and have the following comments:

**Model Registry UI**
- Why is the “Starting” in purple on the 6th line? The others are in gray.

**Model transfer jobs**
- These labels should match the colors in the figma
    - Failed: #b1380b
    - Complete: #3d7317
    - Running: #0066cc

**Model transfer jobs modal**
- Failed: Use #b1380b for both the outline and fa-exclamation-circle icon

**Comment by rsun19** [comment]
@Misjohns updated, here are the new screenshots:

Screenshot 2026-04-06 at 2 51 32 PM
Screenshot 2026-04-06 at 2 51 27 PM
Screenshot 2026-04-06 at 2 50 36 PM
Screenshot 2026-04-06 at 2 50 27 PM
Screenshot 2026-04-06 at 2 50 16 PM

**Comment by Misjohns** [comment]
LGTM Thank you @rsun19

**Comment by rsun19** [comment]
cc @christianvogt @lucferbux @jenny-s51

**Comment by christianvogt** [comment]
We need to validate the material UI components function with these changes and if we need any theme fixes.
@jenny-s51 can help with guiding theme changes if necessary

**Comment by rsun19** [comment]
Blocked by https://github.com/kubeflow/model-registry/pull/2598

**Comment by manaswinidas** [comment]
#2598 is merged

**Comment by lucferbux** [comment]
@jenny-s51 are those changes validated, I've run it after rebasing with main and everything looks fine, but I don't have full context, can you take a look and if everything looks great we can lgtm and approve

**Comment by rsun19** [comment]
Screenshot 2026-04-27 at 9 42 13 AM

**Review by jenny-s51 — APPROVED** [review]
Thanks @rsun19 - LGTM, verified in the MR hub UI - labels look and work as expected with the bump to v1.15.4 of the mod arch library packages.  🎉 

This PR is ready to approve at your convenience @lucferbux - thank you 🙏
